Woof/Woof/ViewModel/LoginViewModel.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+/// An interface responsible for changing the current user role in the app on the `LoginView`.
+final class LoginViewModel: ObservableObject {
+    /// Requests model layer to save selected user role.
+    ///
+    /// - Parameter userRole: The role to be saved.
+    /// - Returns: A boolean indicating whether the operation was successful.
+    func set(_ userRole: Role) -> Bool {
```

Comment:
Will we use the `Bool` result in the app somehow for now?

Reply:
no

Reply:
I'll remove it

Reply:
we use this in tests

Reply:
So you decide to return something just for this test?

```swift
func testReturnsTrueWhenSetEveryPossibleRole() {
        Role.allCases.forEach {
            XCTAssertTrue(testLoginViewModel.set($0))
        }
 }
```

You can create a test for each case and verify that the value was saved, which will mean that our operation was successful as you made it for the override test case.

Reply:
Then maybe it makes sense to use `@discardableResult`?

Reply:
> You can create a test for each case and verify that the value was saved, which will mean that our operation was successful as you made it for the override test case.

ok I'll rewrite this test